presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1153 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.ACCUMULO_TABLE_DNE;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.ACCUMULO_TABLE_EXISTS;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.COLUMN_NOT_FOUND;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.INTERNAL_ERROR;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.MINI_ACCUMULO;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.NOT_SUPPORTED;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.UNEXPECTED_ACCUMULO_ERROR;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VALIDATION;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VIEW_ALREADY_EXISTS;
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VIEW_IS_TABLE;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is the main access point for the Presto connector to interact with Accumulo. It is
+ * responsible for creating tables, dropping tables, retrieving table metadata, and getting the
+ * ConnectorSplits from a table.
+ * <p>
+ * Classes that requires an Accumulo Connector object should use the static method to retrieve it.
+ * This function will create a MiniAccumuloCluster connection for testing, or a 'real' one for production use.
+ */
+public class AccumuloClient
+{
+    public static final String DUMMY_LOCATION = "localhost:9997";
+
+    private static final Logger LOG = Logger.get(AccumuloClient.class);
+    private static final String MAC_PASSWORD = "secret";
+    private static final String MAC_USER = "root";
+    private static final Splitter COMMA_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
+
+    private static Connector conn = null;
+
+    private final AccumuloConfig conf;
+    private final AccumuloMetadataManager metaManager;
+    private final Authorizations auths;
+    private final Random random = new Random();
+    private final AccumuloTableManager tableManager;
+    private final IndexLookup sIndexLookup;
+
+    /**
+     * Gets an Accumulo Connector based on the given configuration.
+     * Will start/stop MiniAccumuloCluster should that be true in the config.
+     *
+     * @param config Accumulo configuration
+     * @return Accumulo connector
+     */
+    public static synchronized Connector getAccumuloConnector(AccumuloConfig config)
+    {
+        if (conn != null) {
+            return conn;
+        }
+
+        requireNonNull(config, "config is null");
+
+        try {
+            if (config.isMiniAccumuloCluster()) {
+                // Create MAC directory
+                File macDir = Files.createTempDirectory("mac-").toFile();
```

Comment:
Please extract spawning `MiniAccumuloCluster` to a separate method since the rest of code is the same regardless if it's mini cluster or actual one.

Shouldn't `MiniAccumuloCluster` instance be shared between connections (`static`) instead of spawning new whenever `getAccumuloConnector` is called?